### PR TITLE
Add new Dockerfile for TileDB-R user image

### PR DESCRIPTION
**Pull request by Shelnutt2**
Add new Dockerfile for TileDB-R user image.

I've uploaded images to: https://hub.docker.com/r/tiledb/tiledb-r

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,10 +43,28 @@ when building the images, e.g.:
...
+
+#### Instructions
+
+1. Install the Docker daemon from https://www.docker.com/community-edition
+
+2. Clone the TileDB-Docker repo and build the images:
+```
+git clone https://github.com/TileDB-Inc/TileDB-Docker
+cd TileDB-Docker
+docker build -t tiledb/tiledb-r r
```
Comment (eddelbuettel) [inline README.md]: I would recommend against this.  No (or few) other guides show basic usage with a build step.  Show 'docker pull' -- the point is Docker is that you don't need to build.

File: README.md [hunk 2]
```diff
@@ -43,10 +43,28 @@ when building the images, e.g.:
...
+There is two images available for [TileDB-R](https://github.com/TileDB-Inc/TileDB-R).
 
-### Instructions
+### User Image
+
+The user image is meant for end users to be able to use TileDB-R.
+
+#### Instructions
+
+1. Install the Docker daemon from https://www.docker.com/community-edition
```
Comment (eddelbuettel) [inline README.md]: Simpler language:   "Install Docker from ..."    Windows or mac users don't need to know about a daemon service.

File: README.md [hunk 3]
```diff
@@ -43,10 +43,28 @@ when building the images, e.g.:
...
 ## TileDB-R
 
-There is an image available for [TileDB-R](https://github.com/TileDB-Inc/TileDB-R)
-which is used only for CI. This image is build and designed to facilitate CI.
+There is two images available for [TileDB-R](https://github.com/TileDB-Inc/TileDB-R).
 
-### Instructions
+### User Image
+
+The user image is meant for end users to be able to use TileDB-R.
```
Comment (eddelbuettel) [inline README.md]: user image -->  deployment image ?

Simpler:

This image enable user to deploy TileDB R.

(Also, we decided to drop the hyphen in the TileDB R repo last week.)

File: README.md [hunk 4]
```diff
@@ -43,10 +43,28 @@ when building the images, e.g.:
...
+docker build -t tiledb/tiledb-r r
+```
+
+### CI Image
+
+The CI image is setup for unit testing via CI.
+
+#### Instructions
 
 1. Install the Docker daemon from https://www.docker.com/community-edition
```
Comment (eddelbuettel) [inline README.md]: As above.  The distinction between the two is good.

File: r/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    && ../bootstrap --prefix=/usr/local --enable-serialization --enable-s3 --enable-verbose \
+    && make -j $(nproc) \
+    && make -C tiledb install \
+    && ldconfig \
+    && cd /home/docker \
+    && rm -rf tiledb
+
+RUN install.r remotes && installGithub.r TileDB-Inc/TileDB-R
+
+CMD ["bash"]
```
Comment (eddelbuettel) [inline r/Dockerfile]: The bash here is a personal preference.  Most Docker users _may_ prefer `CMD ["R"]`.

**Review by eddelbuettel — APPROVED** [review]
Great to have that.  I have a few nits to pick in the write up, apply as you see fit.

File: README.md (hunk 2 above)
Comment (Shelnutt2) [reply README.md]: I agree, I've removed this entire point (from all sections). Installing docker is outside the scope of this repo, the readme needs an overall.

File: r/Dockerfile (hunk 5 above)
Comment (Shelnutt2) [reply r/Dockerfile]: We can do whatever you feel is appropriate for R users. Shall I change it to `R`?

File: README.md [hunk 6]
```diff
@@ -2,24 +2,49 @@
...
+docker pull tiledb/tiledb-r
+docker run -it --rm tiledb/tiledb-r
+```
 
-### Instructions
+## TileDB R CI Images
 
-1. Install the Docker daemon from https://www.docker.com/community-edition
+For TileDB R a docker images is used for CI. This is available via
+[tiledb/ci-r](https://hub.docker.com/r/tiledb/ci-r).
```
Comment (eddelbuettel) [inline README.md]: The CI image is for us, mostly.  I would not mention it here.

File: README.md [hunk 7]
```diff
@@ -34,23 +59,23 @@ To run:
...
 
-### Instructions
+```
+git clone https://github.com/TileDB-Inc/TileDB-Docker
+cd TileDB-Docker
+docker build -t tiledb/tiledb-r r
+```
 
-1. Install the Docker daemon from https://www.docker.com/community-edition
+### TileDB R CI Image
```
Comment (eddelbuettel) [inline README.md]: Already mentioned on line 29.  Likely one of the two is redundant.

**Comment by ihnorton** [comment]
R doesn't seem to like being run non-interactively for the test step: https://stackoverflow.com/questions/51142381/docker-compose-r-server-fatal-error-you-must-specify-save-no-save

**Comment by eddelbuettel** [comment]
@ihnorton happy to take that question elsewhere. That is _standard and defined R behavior_ so the answer there is correct (I only glanced at itl but I recently have answered a very similar one).  In brief we have, _inter alia_,

```sh
edd@rob:~$ echo 2+2 | R -s
[1] 4
edd@rob:~$ R -s -e '2+2'
[1] 4
edd@rob:~$ 

```

as well the dedicated front-end `Rscript` (as well as the alternate `r` [Disclaimer: I'm behind that]).

**Comment by eddelbuettel** [comment]
@ihnorton And to be even more explicit, retaking the above example:

```sh
edd@rob:~$ echo "2+2" | R -s
[1] 4
edd@rob:~$ echo "2+2" | R 
Fatal error: you must specify '--save', '--no-save' or '--vanilla'
edd@rob:~$ 
```

R is first and foremost an _interactive_ environment and shell.  The error message from the SO question is likely due to a simple misunderstanding: you _can_ pipe into R but you then _have to_ turn on a switch such as `-s`.  Or else you get that error.  Documented and long-established behavior.

**Comment by ihnorton** [comment]
That's fine, I'm just looking for a fix for the docker failure.

**Comment by eddelbuettel** [comment]
There must be something wrong with the setup in the Dockerfile.  Maybe a variable a empty.  If you expand the failing run it ends on (apologies for image but this stuff looses formatting when copied and pasted)

image

I can simulate that here (on my box) with another container (using our official R container):

```sh
edd@rob:~$ docker run r-base
Fatal error: you must specify '--save', '--no-save' or '--vanilla'
edd@rob:~$ 
```

Fails the same way because it invokes the (interactive) R the same way without a terminal.  Adding a terminal helps:

```sh
edd@rob:~$ docker run -ti r-base

R version 4.0.0 (2020-04-24) -- "Arbor Day"
Copyright (C) 2020 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> cat("hello, world\n")
hello, world
> q()
Save workspace image? [y/n/c]: n
edd@rob:~$ 
```

Maybe you wanted something like `docker run --rm -ti -v$(PWD):/work -w /work R CMD check  .` ?
Did you mean to check the pa

**Comment by ihnorton** [comment]
Repo no longer in use.

**Pull request closed without merging**